### PR TITLE
Infinite scroll

### LINKED ISSUE #2: Infinite scroll of plays on the home page
Right now I just display the latest play for each sport on the Home page. Instead show the latest N, and allow the user to infinite scroll through all of them.

### PR DESCRIPTION
**Pull request by greg-finley**
Closes https://github.com/greg-finley/alphabet-game-app/issues/2

We actually have all the play JSON in hand when the app loads. But I think infinite scroll is good because we grab the player images to make each card and don't just dump all of them in their browser. It doesn't take any time really to load the next set.

### CODE REVIEW

File: src/components/ScoreboardCard.tsx [hunk 1]
```diff
@@ -35,7 +35,20 @@ export default function ScoreboardCard(props: ScoreboardCardProps) {
...
+                  <img
+                    style={{
+                      width: "100%",
+                      height: "100%",
+                      objectFit: "cover",
+                    }}
+                    src="https://a.espncdn.com/combiner/i?img=/i/headshots/nophoto.png&h=110&w=110&scale=crop"
+                    alt={play.player_name}
+                  />
+                }
```
Comment (garrett-thompson) [inline src/components/ScoreboardCard.tsx]: This is weird. Why the curly braces? Why not pass it as a named prop like `fallbackImg`?

File: src/components/MostRecentScoreboard.tsx [hunk 2]
```diff
@@ -2,25 +2,46 @@ import * as React from "react";
...
+    setItems(
+      items.concat(
+        plays
+          .slice(items.length, items.length + 6)
+          .map((play, i) => (
+            <ScoreboardCard play={play} key={i + items.length} />
+          ))
+      )
+    );
+  };
```
Comment (greg-finley) [inline src/components/MostRecentScoreboard.tsx:28]: Slice and map two different ways feels redundant, but I do want to initialize to first six elements. Could be a better way.

**Review by garrett-thompson — COMMENTED** [review]
If you already have the play data and you just want to defer loading the images, you should be able to tag them with `loading="lazy"` like this describes: https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading#images_and_iframes

I don't think any of the other code is necessary in that case.

File: src/components/ScoreboardCard.tsx (hunk 1 above)
Comment (greg-finley) [reply src/components/ScoreboardCard.tsx]: Was following this: https://stackoverflow.com/questions/63276812/how-to-change-with-fallback-image-with-material-ui. i.e. Avatar falls back to its children, you can't pass a fallback as a prop.

File: src/components/MostRecentScoreboard.tsx (hunk 2 above)
Comment (garrett-thompson) [reply src/components/MostRecentScoreboard.tsx:28]: Eh it's fine. You could always resort to a for loop if you were _really_ worried about it, but you shouldn't be.

```js
let itemsToshow = [];
for (let i = 0; i < items.length; i++) {
  const item = <ScoreboardCard play={plays[i]} key={i} />;
  itemsToShow.push(item);
}
// untested, not sure if works
```

**Comment by greg-finley** [comment]
> If you already have the play data and you just want to defer loading the images, you should be able to tag them with `loading="lazy"` like this describes: https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading#images_and_iframes
> 
> I don't think any of the other code is necessary in that case.

It's ok to just dump all the cards on them then, without paginating it?

File: src/components/ScoreboardCard.tsx (hunk 1 above)
Comment (garrett-thompson) [reply src/components/ScoreboardCard.tsx]: I see. Then you can just drop the surrounding curlys on 39 & 50

**Comment by garrett-thompson** [comment]
> It's ok to just dump all the cards on them then, without paginating it?

Yea, since lazy loading will fix the network request issue. 

The only other concern you'll _eventually_ run into is number of objects you're rendering in the DOM with your initial data. It can get slow if you render a ton of stuff. But when you hit that limit, you then use a virtualized list.

**Comment by greg-finley** [comment]
> > It's ok to just dump all the cards on them then, without paginating it?
> 
> Yea, since lazy loading will fix the network request issue.
> 
> The only other concern you'll _eventually_ run into is number of objects you're rendering in the DOM with your initial data. It can get slow if you render a ton of stuff. But when you hit that limit, you then use a virtualized list.

Hmmm ... It doesn't seem like Material UI Avatar supports lazy loading directly. Should I wrap in react-lazyload?

**Comment by garrett-thompson** [comment]
Ohh. No, that's for loading the code into the browser lazily. It splits your bundle. It's not related to fetching images in the browser. 

Just stick with current implementation.

**Pull request merged**